src/main/java/io/cloudsoft/terraform/TerraformSshDriver.java

```diff
@@ -151,20 +156,38 @@ public void customize() {
                                 "if grep -q \"No errors detected\" <<< $(unzip -t "+ cfgPath +" ); then "
                                         + "mv " + cfgPath + " " + cfgPath + ".zip && cd " + getRunDir() + " &&"
                                          + "unzip " + cfgPath + ".zip ; fi")
+                        .requiringExitCodeZero()
                         .environmentVariables(getShellEnvironment())
                         .summary("Preparing configuration (unzip of necessary)...")
-                        .requiringExitCodeZero().newTask()
+                        .newTask()
                         .asTask())
                 .add(SshTasks.newSshExecTaskFactory(getMachine(), initCommand())
+                        .requiringZeroAndReturningStdout()
                         .environmentVariables(getShellEnvironment())
                         .summary("Initializing terraform infrastructure")
-                        .requiringZeroAndReturningStdout().newTask()
+                        .newTask()
                         .asTask())
                 .build());
         DynamicTasks.waitForLast();
 
-        if (initTask.asTask().isError()) {
-            throw new IllegalStateException("Error executing `terraform init`! ");
+        try {
+            final StringBuilder message = new StringBuilder();
+            List<Object> result = (List<Object>) initTask.get();
+            for(Object o : result) {
+                if(!(o instanceof Integer)) {
+                    message.append(o.toString());
+                }
+            }
+            DynamicTasks.queue(
+                    Tasks.create("Verifying Terraform Workspace", () -> {
+                        if(message.toString().contains(EMPTY_TF_CFG_WARN)) {
```

Comment:
A reminder here: we might want to check for "error" level messages from previous output.

Reply:
@algairim I don't understand you comment, what do you think this task is doing?  :)

Reply:
@algairim You are confused of the context here. The `init` command, never fails, it only prints a message if there are no terraform configuration files. The validation of those files is done in the subsequent plan task, which is never reached now.